server/src/main/java/com/cloud/server/StatsCollector.java

```diff
@@ -1533,7 +1536,7 @@ protected void writeBatches(InfluxDB influxDbConnection, String dbName, List<Poi
      * The considered disk stats are the following: bytes read, bytes write, IO read,  and IO write.
      */
     protected boolean isCurrentVmDiskStatsDifferentFromPrevious(VmDiskStatisticsVO previousVmDiskStats, VmDiskStatisticsVO currentVmDiskStats) {
-        if (previousVmDiskStats != null) {
```

Comment:
currentVmDiskStats might be null as well.

Reply:
@GabrielBrascher it looks currentVmDiskStats is not null


https://github.com/apache/cloudstack/blob/9a1486fdebc3d4cfd17d581ec1b66fbd9c3a4bcd/server/src/main/java/com/cloud/server/StatsCollector.java#L735-L746

Reply:
Just updated the code addressing your comment, @ustcweizhou. Thanks for the heads up :+1: